content/blog/2019/12/2019-12-07-generic-webhook-trigger-plugin.adoc

```diff
@@ -0,0 +1,122 @@
+ ** REPO_2
+ * PROJ_2
+ ** REPO_3
+
+Lets think about some features we want to provide for these repositories:
+
+ * Pull request verification
+ * Building snapshot (or pre release if you will)
+ * Building releases
+
+If the developers are use to the repositories being organized like this in Bitbucket, should we not organize them the same way in Jenkins? And if they browse Jenkins should they not find one job per feature, like `pull-request`, `snapshot` and `release`? Each job with parameters only relevant for that feature. I think so! Like this:
+
+* `/` - Jenkins root
+** `/PROJ_1` - A folder, lists git repositories
+*** `/PROJ_1/REPO_1` - A folder, lists jobs relevant for that repo.
+*** `/PROJ_1/REPO_1/release` - A job, performs releases.
+*** `/PROJ_1/REPO_1/snapshot` - A job, performs snapshot releases.
+*** `/PROJ_1/REPO_1/pull-request` - A job, verifies pull requests.
+* `...`
+
+In this example, both `snapshot` and `release` jobs might work with the same git branch. The difference is the feature they provide. Their parameters can be well documented as you don't have to mix parameters relevant for releases and those relevant for snapshots. This cannot be done with https://plugins.jenkins.io/workflow-multibranch[Multibranch Pipeline Plugin] where you specify parameters as `properties` per branch.
+
+
+== Documentation ==
+
+Webhooks are often well documented in the services providing them. See:
+
+- https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html[Bitbucket Cloud]
+- https://confluence.atlassian.com/bitbucketserver/managing-webhooks-in-bitbucket-server-938025878.html[Bitbucket Server]
+- https://developer.github.com/webhooks/[GitHub]
+- https://docs.gitlab.com/ce/user/project/integrations/webhooks.html[GitLab]
+- https://gogs.io/docs/features/webhook[Gogs] and https://docs.gitea.io/en-us/webhooks/[Gitea]
+- https://blog.assembla.com/AssemblaBlog/tabid/12618/bid/107614/Assembla-Bigplans-Integration-How-To.aspx[Assembla]
+- https://developer.atlassian.com/server/jira/platform/webhooks/[Jira]
+
+It bothered me that, even if I understood these webhooks, I was unable to use them. Because I needed to perform development in the plugin I was using in order to provide whatever value from the webhook to the build. That process could take months from PR to actual release. Such a simple thing should really not be an issue.
+
+= The Solution =
+
+My solution is pretty much *back to basics*: We have an automation server (Jenkins) and we want to trigger it on external webhooks. We want to gather information from that webhook and provide it to our build. I call it https://github.com/jenkinsci/generic-webhook-trigger-plugin/[Generic Webhook Trigger Plugin].
+
+The latest docs are available in https://github.com/jenkinsci/generic-webhook-trigger-plugin/[the repo] and I also have a fully working example with GitLab implemented using https://github.com/tomasbjerre/jenkins-configuration-as-code-sandbox[Configuration As Code].
```

Comment:
Confusing link to https://github.com/tomasbjerre/jenkins-configuration-as-code-sandbox . 
```suggestion
The latest docs are available in https://github.com/jenkinsci/generic-webhook-trigger-plugin/[the repo] and I also have a fully working example with GitLab implemented using configuration-as-code. See the repository https://github.com/tomasbjerre/jenkins-configuration-as-code-sandbox[here].
```

Reply:
Can you explain the confusion? Should I remove the sentence or svould I change the link to the point at job dsl configuring the plugin?